### ISSUE TITLE
ingress can not find new apps backends

### ISSUE THREAD

**Issue by MikeMichel** [haproxytech/kubernetes-ingress#58]
When I add a new app to the running ingress it is not added to the haproxy.cfg:

```
2019/09/06 13:41:57 controller.go:278 11: backend default-app5-80 does not exist
2019/09/06 13:41:57 controller-haproxy.go:151: 14: ERR transactionId=f249fd2b-4c5c-4873-8206-1fe036cb8a1a 
msg="Proxy 'http': unable to find required use_backend: 'default-app5-80'."
msg="Proxy 'https': unable to find required use_backend: 'default-app5-80'."
2019/09/06 13:41:57 controller-monitor.go:121: 14: ERR transactionId=f249fd2b-4c5c-4873-8206-1fe036cb8a1a 
msg="Proxy 'http': unable to find required use_backend: 'default-app5-80'."
msg="Proxy 'https': unable to find required use_backend: 'default-app5-80'."
```
after restarting the ingress it's fine.

I am applying the app

```
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    run: app5
  name: app5
spec:
  replicas: 1
  selector:
    matchLabels:
      run: app5
  template:
    metadata:
      labels:
        run: app5
    spec:
      containers:
      - name: app5
        image: httpd
        ports:
        - containerPort: 80

---
apiVersion: v1
kind: Service
metadata:
  labels:
    run: app5
  name: app5
  annotations:
    haproxy.org/check: "enabled"
    haproxy.org/forwarded-for: "enabled"
    kubernetes.io/ingress.class: "haproxy"
spec:
  selector:
    run: app5
  ports:
  - name: port-1
    port: 80
    protocol: TCP
    targetPort: 80
```

and then I update the ingress by adding another host entry.
```
.....
  - host: bla.example.de
    http:
      paths:
      - backend:
          serviceName: app5
          servicePort: 80
        path: /
....
```

I am using v1.2.1

Or is it mandatory to give every app it's own ingress?

**Comment by oktalz** [comment]
hi @MikeMichel, thx for reporting this.

this was a bug, i fixed it in commit e98569d,
probably today or tomorrow I will tag a new release

**Comment by oktalz** [comment]
new version if 1.2.2 is up with included fix
